src/init/init.c

```diff
@@ -87,7 +87,24 @@ static inline void lp_sram_unpack(void)
 
 #if CONFIG_MULTICORE
 
-#ifndef __ZEPHYR__
+#ifdef __ZEPHYR__
+
+static bool check_restore(void)
```

Comment:
sorry, I didn't say this explicitly - you have to change this function's prototype for non-Zephyr case in line 109 below too then